### PR TITLE
use emplace for uninit data

### PR DESCRIPTION
**Pull request by 9il**
follow up fix for #93

### CODE REVIEW

**Comment by 9il** [comment]
ping @drug007

File: source/asdf/serialization.d [hunk 1]
```diff
@@ -1984,29 +1984,33 @@ void deserializeValue(V : T[], T)(Asdf data, ref V value)
...
+				// create array of uninitialized elements
+				// and initialize them using static `deserialize`
+
+				import std.array : uninitializedArray;
+				value = (()@trusted => uninitializedArray!(T[])(elems.save.count))();
 				foreach(ref e; value)
 				{
-					.deserializeValue(elems.front, e);
+					import std.conv: emplace;
+					cast(void)(()@trusted => emplace(&e, T.deserialize(elems.front)))();
```
Comment (9il) [inline source/asdf/serialization.d]: D has eleborate assign, so it not safe to assign the data to unitialiazed values

Comment (9il) [reply source/asdf/serialization.d]: Also, this(this) can be disabled too, so assign may not work at all

Comment (drug007) [reply source/asdf/serialization.d]: Ok, I'll take a look at these issues later (I'll be busy for month about). More probably I'll make these two cases (elaborate assign and disable postblit) explicitly denied to compile. Also I'm not sure that assigning takes place here - `deserialize` is static method that just returns properly constructed instance of aggregate.

Comment (9il) [reply source/asdf/serialization.d]: `e = T.deserialize(elems.front);`

**Pull request merged**